File: CS/WinSolution.Module.Win/MyDatePropertyEditor.cs
using System;
using DevExpress.ExpressApp.Editors;
using DevExpress.ExpressApp.Win.Editors;
using DevExpress.ExpressApp;
using DevExpress.XtraEditors;

namespace WinSolution.Module.Win {
    [PropertyEditor(typeof(DateTime))]
    public class MyDatePropertyEditor : DatePropertyEditor {
        public MyDatePropertyEditor(Type objectType, DictionaryNode info) : base(objectType, info) { }
        protected override object CreateControlCore() {
            DateEdit control = (DateEdit)base.CreateControlCore();
            SetupRepositoryItemDateTimeEdit((RepositoryItemDateTimeEdit)control.Properties);
            return control;
        }
        protected override void SetupRepositoryItem(DevExpress.XtraEditors.Repository.RepositoryItem item) {
            base.SetupRepositoryItem(item);
            SetupRepositoryItemDateTimeEdit((RepositoryItemDateTimeEdit)item);
        }
        protected virtual void SetupRepositoryItemDateTimeEdit(RepositoryItemDateTimeEdit item) {
            item.VistaDisplayMode = DevExpress.Utils.DefaultBoolean.True;
            item.VistaEditTime = DevExpress.Utils.DefaultBoolean.True;
        }
    }
}

File: VB/WinSolution.Module.Win/MyDatePropertyEditor.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.ExpressApp.Editors
Imports DevExpress.ExpressApp.Win.Editors
Imports DevExpress.ExpressApp
Imports DevExpress.XtraEditors

Namespace WinSolution.Module.Win
	<PropertyEditor(GetType(DateTime))> _
	Public Class MyDatePropertyEditor
		Inherits DatePropertyEditor
		Public Sub New(ByVal objectType As Type, ByVal info As DictionaryNode)
			MyBase.New(objectType, info)
		End Sub
		Protected Overrides Function CreateControlCore() As Object
			Dim control As DateEdit = CType(MyBase.CreateControlCore(), DateEdit)
			SetupRepositoryItemDateTimeEdit(CType(control.Properties, RepositoryItemDateTimeEdit))
			Return control
		End Function
		Protected Overrides Sub SetupRepositoryItem(ByVal item As DevExpress.XtraEditors.Repository.RepositoryItem)
			MyBase.SetupRepositoryItem(item)
			SetupRepositoryItemDateTimeEdit(CType(item, RepositoryItemDateTimeEdit))
		End Sub
		Protected Overridable Sub SetupRepositoryItemDateTimeEdit(ByVal item As RepositoryItemDateTimeEdit)
			item.VistaDisplayMode = DevExpress.Utils.DefaultBoolean.True
			item.VistaEditTime = DevExpress.Utils.DefaultBoolean.True
		End Sub
	End Class
End Namespace
